### ISSUE TITLE
[[80] Sneak Preview (13351)] - Location have kill markers instead of the  magnifying glass

### ISSUE THREAD

**Issue by Cryces** [Questie/Questie#4367]
![image](https://user-images.githubusercontent.com/2063902/194709970-287a19b5-ec69-4015-a9f7-0e4fb81332fe.png)

**Comment by Yttrium-tYcLief** [comment]
This is a multi-layered issue with the way the database works and the way objectives are set up in the game itself.

Internally within Questie, we have two ways of displaying objective markers: Objectives and extraObjectives. Objectives are the primary objectives of the quest. These can be "monster", "object", "item", "reputation", "killCredit", or "player". It's possible there are others but those are the core ones to talk about here. For those objectives, Questie *only* uses certain icons:

- For `monsterObjective`, Questie displays a slay icon (sword) at the known coordinates for the monster
- For `objectObjective`, Questie displays an object icon (gear) at the known coordinates for the object
- For `itemObjective`, Questie checks to see how that item is obtained; it flags monsters that drop the item with a slay icon (sword), and objects that contain the item with an object icon (gear)
- For `reputationObjective`, Questie doesn't display anything
- For `killCreditObjective`, Questie displays a slay icon (sword) at the known coordinates for the eligible monsters
- For `playerObjective`, Questie actually does not understand this objective type (yet)

extraObjectives are basically manual icons that we can place at any point; we can target NPCs, objects, raw coordinates, items, etc. We choose what custom text is used for the label, and we can choose what icon to display. If you ever see an event (magnifying glass) or talk (speech bubble) icon, these are guaranteed to be extraObjectives.

As an example: Let's say we have a quest where you have to go to a certain location, talk to an NPC, and then an enemy spawns, and you have to kill the enemy. The quest objective might simply be a `monsterObjective` to kill the spawned enemy mob. However, the NPC you have to interact with is not marked, so Questie's objective icons for this quest can seem a bit unintuitive to users. In this situation, we typically add an extraObjective targeting the NPC you have to interact with to summon the mob, and we use a text label indicating as such.

*However*, certain quests structured this exact same way may have their underlying Objectives laid out a bit differently; they may have one objective for "0/1 Spoken to NPC", and then a separate objective for "0/1 Monster slain". While these are clearly different types of objectives, under the hood, they're *both* monsterObjectives, so Questie draws a slay icon for both.

This specific quest is just an area discovery quest, but it targets four NPCs which do not exist; for instance, [South Gate KC Bunny](https://www.wowhead.com/wotlk/npc=32195/south-gate-kc-bunny). When you walk to the South Gate, the game credits you with a killcredit for the South Gate KC Bunny, and then the quest shows you finished that objective.

TL;DR:
**We currently have no mechanism for Questie to override the icons of specific objectives with custom ones.** Because of this, this is actually more of a feature request than a data correction, as no amount of database tweaking could achieve the desired outcome here.

**Comment by Yttrium-tYcLief** [comment]
#4439 will not directly fix the quest mentioned, but we're exploring options through that fix that may affect this quest in the future.
